src/main/java/com/amazon/ion/impl/bin/IonEncoder_1_1.java

```diff
@@ -199,10 +198,8 @@ public static int writeDecimalValue(WriteBuffer buffer, final BigDecimal value)
         if (numCoefficientBytes > 0) {
             if (coefficientBytes != null) {
                 buffer.writeFixedIntOrUInt(coefficientBytes);
-            } else if (numCoefficientBytes == 1){
-                buffer.writeByte((byte) 0);
             } else {
-                throw new IllegalStateException("Unreachable! coefficientBytes should not be null when numCoefficientBytes > 1");
```

Comment:
🗺️ So, this is kind of interesting. I had already determined that this _should_ be unreachable (hence the message in the exception), but IntelliJ deduced that it is unreachable, which means it must be _provably_ unreachable... so I removed it.